src/test/java/nextstep/courses/domian/SessionTest.java

```diff
@@ -0,0 +1,118 @@
+package nextstep.courses.domian;
+
+import nextstep.courses.domain.Amount;
+import nextstep.courses.domain.CoverImage;
+import nextstep.courses.domain.CoverImageFileName;
+import nextstep.courses.domain.CoverImagePixel;
+import nextstep.courses.domain.CoverImageSize;
+import nextstep.courses.domain.Session;
+import nextstep.courses.domain.SessionDuration;
+import nextstep.courses.domain.SessionEnrolment;
+import nextstep.courses.domain.SessionStatusType;
+import nextstep.courses.domain.SessionStudent;
+import nextstep.courses.domain.Students;
+import nextstep.users.domain.NsUser;
+import nextstep.users.domain.NsUserTest;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+class SessionTest {
+
+    @Test
+    @DisplayName("무료 강의라면 인원수 제한없이 신청이 가능하다.")
+    void enrolment_무료() {
+        Students students = new Students(new ArrayList<>(List.of(NsUserTest.SANJIGI, NsUserTest.JAVAJIGI)));
+        SessionStudent sessionStudent = new SessionStudent(students);
+        Amount amount = new Amount();
+        NsUser newUser = new NsUser(3L, "test", "test", "test", "test");
+        SessionEnrolment sessionEnrolment = new SessionEnrolment(sessionStudent, SessionStatusType.RECRUITMENT, amount, true);
+        SessionDuration sessionDuration = new SessionDuration(LocalDateTime.now().plusDays(1), LocalDateTime.now().plusDays(3));
+        CoverImage coverImage = new CoverImage(new CoverImageFileName("test.png"), new CoverImageSize(300), new CoverImagePixel(300, 200));
+
+        Session session = new Session(1L, sessionDuration, sessionEnrolment, coverImage);
+        session.enrolment(newUser, 0L);
+
+        int actual = students.totalCount();
+        int expected = 3;
+
+        assertThat(actual).isEqualTo(expected);
+    }
+
+    @Test
+    @DisplayName("유료 강의라면 인원수 인원수에 맞게만 신청이 가능하다.")
+    void enrolment_유료() {
+        Students students = new Students(new ArrayList<>(List.of(NsUserTest.SANJIGI, NsUserTest.JAVAJIGI)));
+        SessionStudent sessionStudent = new SessionStudent(students, 3);
+        Amount amount = new Amount(30000L);
+        NsUser newUser = new NsUser(3L, "test", "test", "test", "test");
+        SessionEnrolment sessionEnrolment = new SessionEnrolment(sessionStudent, SessionStatusType.RECRUITMENT, amount, false);
+        SessionDuration sessionDuration = new SessionDuration(LocalDateTime.now().plusDays(1), LocalDateTime.now().plusDays(3));
+        CoverImage coverImage = new CoverImage(new CoverImageFileName("test.png"), new CoverImageSize(300), new CoverImagePixel(300, 200));
+
+        Session session = new Session(1L, sessionDuration, sessionEnrolment, coverImage);
```

Comment:
> 추가로 Session 같은 경우 내부에서 여러 객체를 나누다 보니 테스트 시에 여러 객체를 생성해야 하는 상황이 생기는데 이를 어떻게 해결할 수 있을까요?

음... 우선 저는 두가지 방법이 떠오르네요 🤔 
- 생성자를 다양하게 활용해보는 방법
- 테스트 픽스처를 활용해보는 방법

```suggestion
        Session session = new Session(Session_기간생성됨(), SessionEnrolment_모집중생성됨(), CoverImage_생성됨());
```

참고 : [테스트 픽스처 올바르게 사용하기](https://jojoldu.tistory.com/611)

Reply:
좋은 참고글 감사합니다 ㅎㅎ 해당 블로그글을 참조하여 최종 수정하였습니다! https://github.com/next-step/java-lms/pull/238/commits/90498983a77a540a5bf3fb14323a71af18d8eb2a